ext/EvoTreesCUDAExt/fit.jl

```diff
@@ -1,267 +1,361 @@
-                    else
-                        pred_leaf_cpu!(tree.pred, n, nodes[n].∑, L, params)
-                    end
+                    EvoTrees.pred_leaf_cpu!(tree.pred, n, node_sum_view, EvoTrees.MAE, params)
                 end
+            else
+                EvoTrees.pred_leaf_cpu!(tree.pred, n, node_sum_view, L, params)
             end
         end
-        n_current = copy(n_next)
-        depth += 1
-    end # end of loop over active ids for a given depth
-
+    else
+        # Standard loss: pred = -g / (h + λ*w + L2)
+        nodes_sum_cpu = Array(cache.nodes_sum_gpu)
+        for n in leaf_nodes
+            node_sum_view = view(nodes_sum_cpu, :, n)
+            EvoTrees.pred_leaf_cpu!(tree.pred, n, node_sum_view, L, params)
+        end
+    end
+    
     return nothing
 end
 
+# Apply splits kernel: decide split vs leaf, compute child gradients/predictions
+# Child gradient sums extracted from histogram enable subtraction at next level
+@kernel function apply_splits_kernel!(
+    tree_split, tree_cond_bin, tree_feat, tree_gain, tree_pred,
+    nodes_sum, n_next, n_next_active,
+    best_gain, best_bin, best_feat, h∇, active_nodes, feattypes,
+    depth, max_depth, lambda, gamma, L2, K_val
+)
+    n_idx = @index(Global)
+    node = active_nodes[n_idx]
+    eps = eltype(tree_pred)(1e-8)
 
-# grow a single oblivious tree - grow through all depth
-function grow_otree!(
-    tree::EvoTrees.Tree{L,K},
-    nodes::Vector{N},
-    params::EvoTrees.EvoTypes,
-    ∇::CuMatrix,
-    js,
-    is,
-    left,
-    right,
-    h∇::CuArray,
-    x_bin::CuMatrix,
-    feattypes::Vector{Bool},
-    monotone_constraints,
-) where {L,K,N}
-
-    # initialize
-    n_current = [1]
-    depth = 1
-    jsg = CuVector(js)
+    @inbounds if depth < max_depth && best_gain[n_idx] > gamma
+        tree_split[node] = true
+        tree_cond_bin[node] = best_bin[n_idx]
+        tree_feat[node] = best_feat[n_idx]
+        tree_gain[node] = best_gain[n_idx]
 
-    # initialize summary stats
-    copyto!(nodes[1].∑, sum(Float64, view(∇, :, nodes[1].is), dims=2))
+        child_l = node << 1
+        child_r = (node << 1) + 1
+        feat = Int(tree_feat[node])
+        bin = Int(tree_cond_bin[node])
+        is_numeric = feattypes[feat]
 
-    # grow while there are remaining active nodes
-    while length(n_current) > 0 && depth <= params.max_depth
-        offset = 0 # identifies breakpoint for each node set within a depth
-        n_next = Int[]
-
-        if depth == params.max_depth
-            for n in n_current
-                if L <: Quantile
-                    pred_leaf_cpu!(tree.pred, n, nodes[n].∑, L, params, ∇, nodes[n].is)
-                else
-                    pred_leaf_cpu!(tree.pred, n, nodes[n].∑, L, params)
+        # Compute child gradient sums from histogram (enables subtraction at next level!)
+        # This is CRITICAL: stores gradient sums so next level can use:
```

Comment:
Why mentions that `child gradient sums`  is necessary for `subtract_hist_kernel!`? The later only depends on `h∇` to perform the subtraction, per: https://github.com/Evovest/EvoTrees.jl/pull/299/files#diff-d696002f188318b01ef653ce29625f5597f4bc2f629f9885226494cc4286fd18R121-R124